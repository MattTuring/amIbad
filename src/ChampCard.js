import React from 'react'


const ChampCard = ({name, amIBad, champId, image, addChampToProps, heart}) => {
  return (
    <article className='container' onClick={() => amIBad(champId, name)}>
    <h4>{name}</h4>
    <img className='champ image' src={image} alt='champ'/>
    <div className="middle">
      <div className="text">
      Am I Bad?
      {addChampToProps && <img src={heart} onClick={() => addChampToProps(name, champId, image, heart)} className='favorite' alt="favorite"/>}
      {!addChampToProps && <img src={heart}  className='favorite' alt="favorite"/>}
      </div>
    </div>
    </article>
  )
}

export default ChampCard
