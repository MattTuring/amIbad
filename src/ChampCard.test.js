import React from 'react';
import { shallow } from 'enzyme';
import ChampCard  from './ChampCard.js';

describe('Form', () => {
  let mockFunction = jest.fn()
  let mockFunction2 = jest.fn()

    const wrapper = shallow(<ChampCard
      name={'Aatrox'}
      image={'string.png'}
      champId={'12'}
      key={'12'}
      amIBad={mockFunction}
      addChampToProps={mockFunction}
      heart={'string.png'}
    />)


  it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
  });

  it('onClick for container', () => {
    wrapper.find('.container').simulate('click');
    expect(mockFunction).toHaveBeenCalled()
  });

  it('on click for heart', () => {
    wrapper.find('.favorite').simulate('click');
    expect(mockFunction).toHaveBeenCalled()
  });

  const wrapper2 = shallow(<ChampCard
    name={'Aatrox'}
    image={'string.png'}
    champId={'12'}
    key={'12'}
    amIBad={mockFunction2}
    heart={'string.png'}
  />)

  it('on click for heart', () => {
    wrapper2.find('.favorite').simulate('click');
    expect(mockFunction2).not.toHaveBeenCalled()
  });


})
