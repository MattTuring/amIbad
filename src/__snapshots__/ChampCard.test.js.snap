// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should match the snapshot 1`] = `
<article
  className="container"
  onClick={[Function]}
>
  <h4>
    Aatrox
  </h4>
  <img
    alt="champ"
    className="champ image"
    src="string.png"
  />
  <div
    className="middle"
  >
    <div
      className="text"
    >
      Am I Bad?
      <img
        alt="favorite"
        className="favorite"
        onClick={[Function]}
        src="string.png"
      />
    </div>
  </div>
</article>
`;
